rust/arrow/src/record_batch.rs

```diff
@@ -245,6 +295,12 @@ impl RecordBatch {
     }
 }
 
+impl Default for RecordBatch {
```

Comment:
Not totally sure about this because it bypasses the invariants of `validate_new_batch` - it seems like we don't want to allow for 0-column record batches to exist. Might be an argument in favour of the `impl From` approach?